app/api-v1/api-doc.js

```diff
@@ -26,10 +26,463 @@ const apiDoc = {
         description: 'An error occurred',
       },
     },
-    schemas: {},
+    schemas: {
+      NewCatalogueItem: {
+        type: 'object',
+        properties: {
+          externalId: {
+            description: 'id of the catalogue-item in an external ERP',
+            allOf: [{ $ref: '#/components/schemas/OnChainLiteral' }],
+          },
+          name: {
+            description: 'Name of the catalogue-item',
+            allOf: [{ $ref: '#/components/schemas/OnChainLiteral' }],
+          },
+          imageAttachmentId: {
+            description: 'Example image of the item uploaded as an attachment',
+            allOf: [{ $ref: '#/components/schemas/ObjectReference' }],
+          },
+          material: {
+            description: 'Primary material of the constructed catalogue-item',
+            allOf: [{ $ref: '#/components/schemas/OnChainLiteral' }],
+          },
+          alloy: {
+            description: 'Primary alloy present in the constructed catalogue-item',
+            allOf: [{ $ref: '#/components/schemas/OnChainLiteral' }],
+          },
+          price: {
+            description: 'Price of the catalogue-item. This information is not stored on-chain.',
+            type: 'string',
+          },
+          requiredCerts: {
+            description: 'Certification requirements ',
+            type: 'array',
+            maxItems: 10,
+            items: {
+              $ref: '#/components/schemas/CertificationRequirement',
+            },
+          },
+          supplier: {
+            description:
+              'Name of the supplier who is contracted to build the catalogue-item. This information is not stored directly on-chain',
+            type: 'string',
+            maxLength: 255,
+          },
+        },
+      },
+      CatalogueItem: {
+        type: 'object',
+        allOf: [{ $ref: '#/components/schemas/NewCatalogueItem' }],
+        properties: {
+          id: {
+            description: 'local id of the catalogue-item',
+            allOf: [{ $ref: '#/components/schemas/ObjectReference' }],
+          },
+          owner: {
+            description:
+              'Name of the OEM who owns the design of the catalogue-item. This information is not stored directly on-chain',
+            type: 'string',
+            maxLength: 255,
+          },
+        },
+      },
+      CertificationRequirement: {
+        type: 'object',
+        properties: {
```

Comment:
for `inteli-demo`, certificates also have a `metadataKey` attribute so we know which metadata key to assign an uploaded cert file to on-chain
https://github.com/digicatapult/inteli-demo/blob/9b2e12866d8abbc7c07df639ba9c800fd4296f58/src/features/partsSlice.js#L4

Reply:
we could generate the required keys though (and guarantee they're unique)

Reply:
I think we can auto-generate this